Replace kw... with kws...

LilithHafner opened the pull request:
EDIT: This is backward, I actually changed `kw` to `kws` (See discussion).

I notice that we use both `kws...` and `kw...` with little apparent pattern. I prefer using one consistently, so I changed almost all occurrences of `kws...` to `kw...`. I chose `kw` because it is already more common.
```
grep -owr 'kws\.\.\.' base stdlib test | wc -l
      77
grep -owr 'kw\.\.\.' base stdlib test | wc -l 
     160
```

Most of this pr comes from the command 
```
grep -rlw 'kws\.\.\.' base stdlib/Sockets stdlib/Test stdlib/Markdown stdlib/REPL stdlib/InteractiveUtils/ stdlib/LinearAlgebra/ stdlib/Logging/ test | xargs sed -I '' 's/kws\.\.\./kw.../'
```
I only dealt with stdlibs that are a part of the julia repo and contain the string `kws...`.

thchr commented on base/floatfuncs.jl:
```diff
@@ -325,7 +325,7 @@ const ≈ = isapprox
 This is equivalent to `!isapprox(x,y)` (see [`isapprox`](@ref)).
 """
-≉(args...; kws...) = !≈(args...; kws...)
+≉(args...; kw...) = !≈(args...; kw...)
```
I find the singular `kw` odd: by the reasoning that `kws` should be `kw`, it seems `args` should be `arg` here for instance. In both cases, however, the absence of 's' suggests the quantity is singular rather than plural (contrary to intention: i.e., the `...` is there to allow multiple arguments).

LilithHafner replied:
Aye, and that is supported by the observation that In a couple of cases I had to leave `kws...` as is because of a subsequent `for kw in kws` loop.

I think it may be a good idea to switch from `kw` to `kws` in most cases. (though keep `kw...` when there can be at most one keyword argument)

I just noticed that [Blue](https://github.com/invenia/BlueStyle) recommends `kwargs...`, an even more verbose option that is quite popular in the stdlibs already:
```
grep -owr 'kwargs\.\.\.' stdlib | wc -l
     499
```
But only slightly more common than `kw` in base:
```
x@X julia_dm % grep -owr 'kwargs\.\.\.' base | wc -l            
      88
x@X julia_dm % grep -owr 'kws\.\.\.' base | wc -l 
      13
x@X julia_dm % grep -owr 'kw\.\.\.' base | wc -l 
      77
```

KristofferC replied:
> I just noticed that [Blue](https://github.com/invenia/BlueStyle) recommends kwargs..., an even more verbose option that is quite popular in the stdlibs already:

I think `kws` is better.

LilithHafner replied:
There seems to be a solid consensus for `kws` instead of `kw` backed by sensible reasoning. I'm going to reverse this PR to change `kw` to `kws`.

Commit:
Use sed to replace kws... with kw...

I ran this command twice:
grep -rlw 'kw\.\.\.' base stdlib/Sockets stdlib/Test stdlib/Markdown stdlib/REPL stdlib/InteractiveUtils/ stdlib/LinearAlgebra/ stdlib/Logging/ test | xargs sed -I '' 's/kw\.\.\./kws.../'

Commit:
Use sed to replace kwarg... with kwargs...

I ran this command once:
grep -rlw 'kwarg\.\.\.' base stdlib/LibGit2/ | xargs sed -I '' 's/kwarg\.\.\./kwargs.../'

LilithHafner commented:
This PR currently changes all occurrences of `kw` to `kws` and all occurrences of `kwarg` to `kwargs`. Should we
1) Use the singular `kw` for cases where there can only be one or zero keyword arguments (I think no, but it could go either way)
2) Change all occurrences of `kwargs` to `kws` (I think yes, at least in base, c.f. https://github.com/JuliaLang/julia/pull/46225#discussion_r939557146)
3) Change all occurrences of `kws` to `kwargs` (I think no)

fredrikekre commented on base/accumulate.jl:
```diff
@@ -269,10 +269,10 @@ julia> accumulate(+, fill(1, 2, 5), dims=2, init=100.0)
-        return collect(Iterators.accumulate(op, A; kw...))
+        return collect(Iterators.accumulate(op, A; kws...))
     end
     nt = values(kw)
```
This will error now. A change like this feels a bit scary since we don't have 100% test coverage. (Personally I think changes like this are rather pointless unless 1) this will be included in a style guide for JuliaLang/julia and 2) will be enforced in code review.)

LilithHafner replied:
1) Yes, I'm looking through `grep -Ir '[^a-z_:0-9]kw[^a-z_0-9]' base stdlib test` now, and then I'll have to look through the cumulative diff too.
2) Fair point, it was bothering me to see inconsistencies, but the best course of action may well be to ignore it. I'll close this PR until someone else is also bothered by the inconsistency.
